cython_args option has no effect

**Describe the bug**
When using Meson to transpile Cython files, compiler arguments specified via `cython_args` under `[built-in options]` in a cross/native file (or configured via `-Dcython_args=...` in `meson configure`/`meson setup`) are not passed to the transpiler. 

**To Reproduce**
`meson.build`
```meson
project('foo', 'c', 'cython')

py = import('python').find_installation()
py.extension_module('t', 't.pyx')
```

`t.pyx`
```pyrex
cimport numpy
```

Create virtualenv with Cython.
```console
$ python -m venv .venv && .venv/bin/pip install cython && source .venv/bin/activate
```

Install NumPy in a separate directory
```console
$ mkdir -p lib && .venv/bin/pip install numpy --target lib
```

Setup build directory
```console
$ meson setup build -Dcython_args="-I$PWD/lib"
```

When building, it fails because it can't find numpy. From the command used, it didn't provide `cython_args`.
```console
$ ninja -C build
ninja: Entering directory `build'
[1/3] Compiling Cython source /tmp/repro/t.pyx
FAILED: [code=1] t.cpython-314-x86_64-linux-gnu.so.p/t.pyx.c 
cython -M --fast-fail -3 /tmp/repro/t.pyx -o t.cpython-314-x86_64-linux-gnu.so.p/t.pyx.c

Error compiling Cython file:
------------------------------------------------------------
...
cimport numpy
        ^
------------------------------------------------------------
/tmp/repro/t.pyx:1:8: 'numpy.pxd' not found
ninja: build stopped: subcommand failed.
```

If it provided the `-I` option specified in `cython_args` it would have builded correctly
```console
$ pushd build; cython -I/tmp/repro/lib -M --fast-fail -3 /tmp/repro/t.pyx -o t.cpython-314-x86_64-linux-gnu.so.p/t.pyx.c; popd
```

Checking `meson configure build` shows the option was parsed:
```
  cython_args                    [-I/tmp/repro/lib]                                                Extra arguments passed to the cython compiler                
```

**Expected behavior**
I expect Meson to pass `cython_args` to the `cython` command in `build.ninja`.

**system parameters**
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Fedora 44

* what Python version are you using e.g. 3.8.0
3.14.6

* what `meson --version`
1.11.1

* what `ninja --version` if it's a Ninja build
1.13.2

* cython version: 3.2.8
* numpy version: 2.5.1

--- a/mesonbuild/backend/ninjabackend.py
+++ b/mesonbuild/backend/ninjabackend.py
@@ -1898,6 +1898,7 @@ class NinjaBackend(backends.Backend):
         args += cython.get_option_std_args(target, target.subproject)
         args += self.build.get_global_args(cython, target.for_machine)
         args += self.build.get_project_args(cython, target)
+        args += self.environment.coredata.get_external_args(target.for_machine, cython.get_language())
         args += target.get_extra_args('cython')
 
         ext = self.get_target_option(target, OptionKey('cython_language', machine=target.for_machine))

--- a/mesonbuild/backend/ninjabackend.py
+++ b/mesonbuild/backend/ninjabackend.py
@@ -1898,7 +1898,10 @@ class NinjaBackend(backends.Backend):
         args += cython.get_option_std_args(target, target.subproject)
         args += self.build.get_global_args(cython, target.for_machine)
         args += self.build.get_project_args(cython, target)
-        args += self.environment.coredata.get_external_args(target.for_machine, cython.get_language())
+        cython_args = self.environment.coredata.optstore.get_value_for(
+            OptionKey('cython_args', target.subproject, target.for_machine))
+        assert isinstance(cython_args, list), 'for mypy'
+        args += cython_args
         args += target.get_extra_args('cython')
 
         ext = self.get_target_option(target, OptionKey('cython_language', machine=target.for_machine))

--- a/mesonbuild/backend/ninjabackend.py
+++ b/mesonbuild/backend/ninjabackend.py
@@ -1898,6 +1898,10 @@ class NinjaBackend(backends.Backend):
         args += cython.get_option_std_args(target, target.subproject)
         args += self.build.get_global_args(cython, target)
         args += self.build.get_project_args(cython, target)
+        cython_args = self.environment.coredata.optstore.get_value_for(
+            OptionKey('cython_args', target.subproject, target.for_machine))
+        assert isinstance(cython_args, list), 'for mypy'
+        args += cython_args
         args += target.get_extra_args('cython')
 
         ext = self.get_target_option(target, OptionKey('cython_language', machine=target.for_machine))

--- a/test cases/cython/3 cython_args/builtincythonargs.pyx
+++ b/test cases/cython/3 cython_args/builtincythonargs.pyx
@@ -1,0 +1,5 @@
+def test():
+    IF BUILTIN_VALUE:
+        return 1
+    ELSE:
+        return 0

--- a/test cases/cython/3 cython_args/meson.build
+++ b/test cases/cython/3 cython_args/meson.build
@@ -24,6 +24,11 @@ mod = python.extension_module(
   ],
 )
 
+builtin_mod = python.extension_module(
+  'builtincythonargs',
+  files('builtincythonargs.pyx'),
+)
+
 test(
   'test',
   python,

--- a/test cases/cython/3 cython_args/test.json
+++ b/test cases/cython/3 cython_args/test.json
@@ -1,0 +1,9 @@
+{
+  "matrix": {
+    "options": {
+      "cython_args": [
+        { "val": "--compile-time-env BUILTIN_VALUE=1" }
+      ]
+    }
+  }
+}

--- a/test cases/cython/3 cython_args/test.py
+++ b/test cases/cython/3 cython_args/test.py
@@ -1,3 +1,5 @@
 import cythonargs
+import builtincythonargs
 
 assert cythonargs.test() == 1
+assert builtincythonargs.test() == 1

